src/ntrip_ping.c

```diff
@@ -152,9 +160,14 @@ int request(void)
     return -1;
   }
 
+  size_t client_id_nbytes = snprintf(NULL, 0, "X-SwiftNav-Client-Id: %s", client_id) + 1; /* +1 for the \0 */
+  char *client_id_header = malloc(client_id_nbytes);
+  snprintf(client_id_header, client_id_nbytes, "X-SwiftNav-Client-Id: %s", client_id);
```

Comment:
I'd just put up a reasonably sized buffer (say 1024) and sprintf into it instead of doing this. This tool is pretty easy going around this stuff.

Reply:
fixed, sorry am pretty scared to use sprintf